fmskill/compare.py

```diff
@@ -1282,19 +1282,29 @@ def __init__(self, observation, modeldata):
         if not isinstance(modeldata, list):
             modeldata = [modeldata]
         for j, data in enumerate(modeldata):
-            df = data.to_dataframe()
+            df = self._model2obs_interp(self.observation, data)
+            if (len(df) > 0) and (len(df) == len(self.observation.df)):
+                ok = self._obs_mod_xy_distance_acceptable(df, self.observation.df)
+                df.iloc[~ok, 1] = np.nan
+                if sum(ok) == 0:
+                    warnings.warn("no (spatial) overlap between model and observation points")
             if j == 0:
-                df[self.obs_name] = observation.df.iloc[:, -1]
                 cols = list(df.columns)
-                cols = list((*cols[0:2], *cols[:1:-1]))
+                cols = list((*cols[:1:-1], *cols[0:2]))
```

Comment:
What are the first two columns?
Is it possible to be more explicit, by referring to the columns by name?

Reply:
I agree with you most of the way. The reason for using column ids is that I don't know what the user is providing in terms of column. But I guess it would be better to change the column names "up front" as the first thing after receiving the input from the user.

Reply:
see https://github.com/DHI/fmskill/pull/45/commits/df05f42b896276c2716be5817227d52755a2754d